### ISSUE TITLE
Error New-NAVContainer

### ISSUE THREAD

**Issue by thenavguy** [Microsoft/navcontainerhelper#373]
Hi, am executing below code and getting error
New-NavContainer -containerName bccu3 -accept_eula -accept_outdated -alwaysPull -assignPremiumPlan `
-auth Windows -enableSymbolLoading `
-imageName mcr.microsoft.com/businesscentral/onprem:cu3-na-ltsc2019 -includeCSide `
-licenseFile C:\Users\softc\Documents\myflf.flf 
-myScripts c:\myfolder -shortcuts Desktop -useBestContainerOS

Error
Creating Windows user softc
Setting SA Password and enabling SA
Program 'SQLCMD.EXE' failed to run: The operation was blocked by parental controlsAt C:\Run\SetupSqlUsers.ps1:1
3 char:5
+     & sqlcmd -S "$databaseServer\$databaseInstance" -Q $sqlcmd
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~.
at <ScriptBlock>, C:\Run\SetupSqlUsers.ps1: line 13
at <ScriptBlock>, C:\Run\navstart.ps1: line 174
at <ScriptBlock>, C:\Run\start.ps1: line 121
at <ScriptBlock>, <No file>: line 1
Initialization of container bccu3 failed
At C:\Program 
Files\WindowsPowerShell\Modules\navcontainerhelper\0.5.0.5\ContainerHandling\Wait-NavContainerReady.ps1:45 
char:17
+ ...             throw "Initialization of container $containerName failed" ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Initialization of container bccu3 failed:String) [], Runtime 
   Exception
    + FullyQualifiedErrorId : Initialization of container bccu3 failed

Output of Docker inspect
[
    {
        "Id": "d6f59addf43f76e0eff4e56e1a317cc694cf3374e6446cbcd96809d476803090",
        "Created": "2019-03-07T00:46:31.5506179Z",
        "Path": "powershell",
        "Args": [
            "-Command",
            "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';",
            ".\\Run\\start.ps1"
        ],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 15996,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2019-03-07T00:46:34.4717333Z",
            "FinishedAt": "0001-01-01T00:00:00Z",
            "Health": {
                "Status": "healthy",
                "FailingStreak": 0,
                "Log": [
                    {
                        "Start": "2019-03-06T19:54:43.855509-05:00",
                        "End": "2019-03-06T19:54:47.9645129-05:00",
                        "ExitCode": 0,
                        "Output": ""
                    },
                    {
                        "Start": "2019-03-06T19:55:18.0142681-05:00",
                        "End": "2019-03-06T19:55:22.2948025-05:00",
                        "ExitCode": 0,
                        "Output": ""
                    },
                    {
                        "Start": "2019-03-06T19:55:52.3358364-05:00",
                        "End": "2019-03-06T19:55:56.8452655-05:00",
                        "ExitCode": 0,
                        "Output": ""
                    },
                    {
                        "Start": "2019-03-06T19:56:26.8937448-05:00",
                        "End": "2019-03-06T19:56:31.1407541-05:00",
                        "ExitCode": 0,
                        "Output": ""
                    },
                    {
                        "Start": "2019-03-06T19:57:01.1902139-05:00",
                        "End": "2019-03-06T19:57:05.8455392-05:00",
                        "ExitCode": 0,
                        "Output": ""
                    }
                ]
            }
        },
        "Image": "sha256:3c94165e2cd0cd8145e28ae5622ad60937b577d13fa31c762c7c154ec0793326",
        "ResolvConfPath": "",
        "HostnamePath": "",
        "HostsPath": "",
        "LogPath": "C:\\ProgramData\\Docker\\containers\\d6f59addf43f76e0eff4e56e1a317cc694cf3374e6446cbcd96809
d476803090\\d6f59addf43f76e0eff4e56e1a317cc694cf3374e6446cbcd96809d476803090-json.log",
        "Name": "/bccu3",
        "RestartCount": 0,
        "Driver": "windowsfilter",
        "Platform": "windows",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": [
                "C:\\ProgramData\\NavContainerHelper:C:\\ProgramData\\NavContainerHelper",
                "C:\\ProgramData\\NavContainerHelper\\Extensions\\bccu3\\my:C:\\Run\\my",
                "C:\\ProgramData\\NavContainerHelper\\Extensions\\bccu3\\Program Files:C:\\navpfiles"
            ],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "default",
            "PortBindings": {},
            "RestartPolicy": {
                "Name": "unless-stopped",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "Dns": [],
            "DnsOptions": [],
            "DnsSearch": [],
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 0,
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "process",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": [],
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": [],
            "DeviceCgroupRules": null,
            "DiskQuota": 0,
            "KernelMemory": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": false,
            "PidsLimit": 0,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": null,
            "ReadonlyPaths": null
        },
        "GraphDriver": {
            "Data": {
                "dir": "C:\\ProgramData\\Docker\\windowsfilter\\d6f59addf43f76e0eff4e56e1a317cc694cf3374e6446cb
cd96809d476803090"
            },
            "Name": "windowsfilter"
        },
        "Mounts": [
            {
                "Type": "bind",
                "Source": "c:\\programdata\\navcontainerhelper",
                "Destination": "c:\\programdata\\navcontainerhelper",
                "Mode": "",
                "RW": true,
                "Propagation": ""
            },
            {
                "Type": "bind",
                "Source": "c:\\programdata\\navcontainerhelper\\extensions\\bccu3\\my",
                "Destination": "c:\\run\\my",
                "Mode": "",
                "RW": true,
                "Propagation": ""
            },
            {
                "Type": "bind",
                "Source": "c:\\programdata\\navcontainerhelper\\extensions\\bccu3\\program files",
                "Destination": "c:\\navpfiles",
                "Mode": "",
                "RW": true,
                "Propagation": ""
            }
        ],
        "Config": {
            "Hostname": "bccu3",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "1433/tcp": {},
                "443/tcp": {},
                "7045/tcp": {},
                "7046/tcp": {},
                "7047/tcp": {},
                "7048/tcp": {},
                "7049/tcp": {},
                "80/tcp": {},
                "8080/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "licenseFile=c:\\run\\my\\license.flf",
                "databaseServer=",
                "ExitOnError=N",
                "locale=en-US",
                "accept_outdated=Y",
                "auth=Windows",
                "databaseInstance=",
                "removePasswordKeyFile=Y",
                "username=softc",
                "enableApiServices=Y",
                "enableSymbolLoading=Y",
                "useSSL=N",
                "securePassword=76492d1116743f0423413b16050a5345MgB8AGIAWABsAHcAdgBCAFoARgBMAEEAVwBkAEcATQBVAFE
AawBPAEIAZgBnAFEAPQA9AHwAYQA5AGMANwA3ADQAOABmADIANABhADgAMQA3ADEANAA4ADMAMgA5ADUANAA5ADkANAAzAGIAZQA3ADEAYQBlAD
EANQAyAGEAOQBhADgAMQA2AGQAZgAwADMAMABmAGIAMAAyADkANABmAGYAOQBkADEAOQBlAGEAOQBkADAAYgA=",
                "passwordKeyFile=c:\\run\\my\\aes.key",
                "accept_eula=Y",
                "DatabaseName=Cronusna",
                "NAVDVDURL=https://nav2016wswe0.blob.core.windows.net/dvd/ae2edd0d-e8ff-4d10-b6a2-2765e31e1c39"
,
                "VSIXURL=https://nav2016wswe0.blob.core.windows.net/dvd/al-2.1.79379.vsix",
                "IsBcSandbox=N",
                "CountryCode=na",
                "COUNTRYURL=https://nav2016wswe0.blob.core.windows.net/dvd/101336e6-4c13-4389-8b28-f23a9639b1c1
"
            ],
            "Cmd": [
                "powershell",
                "-Command",
                "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';",
                ".\\Run\\start.ps1"
            ],
            "Healthcheck": {
                "Test": [
                    "CMD",
                    "powershell",
                    ".\\Run\\HealthCheck.ps1"
                ],
                "Interval": 30000000000,
                "Timeout": 10000000000
            },
            "ArgsEscaped": true,
            "Image": "mcr.microsoft.com/businesscentral/onprem:cu3-na-ltsc2019",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": {
                "country": "na",
                "created": "201901171538",
                "cu": "",
                "eula": "https://go.microsoft.com/fwlink/?linkid=861843",
                "legal": "http://go.microsoft.com/fwlink/?LinkId=837447",
                "maintainer": "Dynamics SMB",
                "nav": "",
                "osversion": "10.0.17763.194",
                "platform": "13.0.27183.0",
                "tag": "0.0.9.0",
                "version": "13.3.27233.0"
            }
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "d6f59addf43f76e0eff4e56e1a317cc694cf3374e6446cbcd96809d476803090",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {
                "1433/tcp": null,
                "443/tcp": null,
                "7045/tcp": null,
                "7046/tcp": null,
                "7047/tcp": null,
                "7048/tcp": null,
                "7049/tcp": null,
                "80/tcp": null,
                "8080/tcp": null
            },
            "SandboxKey": "d6f59addf43f76e0eff4e56e1a317cc694cf3374e6446cbcd96809d476803090",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "nat": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "658ce1c8d76b453f96475a690375b6559657368a7fe7299698ebedc7873a06c8",
                    "EndpointID": "22ca10b81bf5626a49bf327216586e796b7983350cc94d013258816a8caa3732",
                    "Gateway": "172.26.208.1",
                    "IPAddress": "172.26.215.189",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "00:15:5d:d8:7e:08",
                    "DriverOpts": null
                }
            }
        }
    }
]

My Environment is  windows 10 Professional Version 1809 (OS Build 17763.316), 16 GB RAM
Navcontainerhelper Version 0.5.0.5

Tried using NavUserPassword authentication but still getting error.

**Comment by freddydk** [comment]
Duplicate of https://github.com/Microsoft/navcontainerhelper/issues/346
I don't know what you have installed on your machine, which does this, but you are the second person.
My guess is some antivirus or like.
Your only option is probably to use hyperv for now.

**Comment by freddydk** [comment]
Actually - you could try one thing.
Download the attached file to your machine and remove the .txt extension (f.ex. c:\temp\SetupSqlUsers.ps1). You might also have to right click the file, select properties and unblock the file. Then start your contianer with -myscripts @("c:\temp\setupsqlusers.ps1")
This will override the script and use powershell instead of sqlcmd in that particular area - then we can see if it comes past that error.
[SetupSqlUsers.ps1.txt](https://github.com/Microsoft/navcontainerhelper/files/2940281/SetupSqlUsers.ps1.txt)

**Comment by freddydk** [comment]
Did you try the modified SetupSqlUsers.ps1?
Would be good to know whether this solves the issue

**Comment by thenavguy** [comment]
Hi Freddy,

My apologies, was travelling so could not try your suggestion.

After following your suggestion, getting below error
NavContainerHelper is version 0.5.0.5
Host is Microsoft Windows 10 Pro - ltsc2019
Docker Client Version is 18.09.2
Docker Server Version is 18.09.2
Pulling image mcr.microsoft.com/businesscentral/onprem:cu3-na-ltsc2019
cu3-na-ltsc2019: Pulling from businesscentral/onprem
Digest: sha256:b3c0a2baafae1fdd2e52a2d7f6ac9d4fa0ef33727c87e64d2e2776f99262d492
Status: Image is up to date for mcr.microsoft.com/businesscentral/onprem:cu3-na-ltsc2019
Using image mcr.microsoft.com/businesscentral/onprem:cu3-na-ltsc2019
Get-Item : Cannot find path 'C:\WINDOWS\system32\-\ServiceTier\program files\Microsoft Dynamics NAV' because it does not exist.
At C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.5.0.5\ContainerHandling\New-NavContainer.ps1:378 char:24
+ ... vversion = (Get-Item -Path "$navDvdPath\ServiceTier\program files\Mic ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\WINDOWS\syst...ft Dynamics NAV:String) [Get-Item], ItemNotFoundException
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetItemCommand

Executed following script
New-NavContainer -containerName bcna3 -accept_eula -accept_outdated -alwaysPull -assignPremiumPlan `
-auth NavUserPassword - Credential soft.c -enableSymbolLoading `
-imageName mcr.microsoft.com/businesscentral/onprem:cu3-na-ltsc2019 -includeCSide `
-licenseFile C:\Users\softc\Documents\myflf.flf 
-myScripts C:\myfolder\SetupSqlUsers.ps1 -shortcuts Desktop -useBestContainerOS

The strange thing is I was able to run NAV containers earlier, and wanted to start with the latest cu, so deleted the earlier container and started with new one started getting error.

My Laptop details
Windows 10 Version 1809 (OS Build 17763.316)
Windows Defender and Windows Defender Firewall is On

I gave a try with uninstalling Docker Desktop and Container features from windows and added them again.

**Comment by fommo** [comment]
Did you solve it? I get the exact same error and tried to follow all guides I can find.
2-3 months ago I had no problems to set up a new BC container, but now with latest version of Docker and New-NavContainer script I get this "Cannot find path" error no matter what I do.

**Comment by NAVspecialist** [comment]
@fommo  - You dont provide any information so it is very hard to help you. 

What exact version /build of OS are you running?
What version of docker? 
How does your New-NAVcontainer look like?

Thanks

**Comment by thenavguy** [comment]
Hi, Fomo

What Authentication method you are using. Try using NAVUserPassword, specify user name without any specialcharacters [.-].

My Environment details
Windows 10 Pro. Version 1809 (OS Build 17763.379)

 Version:           18.09.2
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        6247962
 Built:             Sun Feb 10 04:12:31 2019
 OS/Arch:           windows/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.2
  API version:      1.39 (minimum version 1.24)
  Go version:       go1.10.6
  Git commit:       6247962
  Built:            Sun Feb 10 04:28:48 2019
  OS/Arch:          windows/amd64
  Experimental:     false

Thanks and Regards,

**Comment by fommo** [comment]
> @fommo - You dont provide any information so it is very hard to help you.
> 
> What exact version /build of OS are you running?
> What version of docker?
> How does your New-NAVcontainer look like?
> 
> Thanks

Well, thanks for replying. It's kind of embarrasing actually. I solved in and I guess that @thenavguy also found it as nothing more was commented here. I wrote here because I got exactly the same error as the last comment in the thread, that the path to "Dynamics NAV" could not be found.
In my case it was trying to find it in a folder "c" under my home folder and I thought it was legit as some temporary folder could look like. But after several tries I found that the fourth parameter was "@credential" instead of variable " $credential". In the New-NAVContainer script the fourth parameter is NavDVDPath, so it defined the path as "c" instead of leaving it empty.
And then I got the same error.
Looking at @thenavguy issue there is a blank space between "-" and "Credential" giving the exact same bevaviour, except that in this case it tries to find the folder "System32-" instead of my "c" folder.
Crazy what a small mistake can do a big difference. :)
